src/streamable/serialize.rs

```diff
@@ -0,0 +1,491 @@
+use super::error::{Error, Result};
```

Comment:
How much of this is ported from `bincode` vs home-grown? If a lot is borrowed, maybe a comment to this effect? "Based on <url to source>"

Reply:
bincode was my main source of inspiration, for sure. I'll add a comment.